py/selenium/webdriver/remote/remote_connection.py

```diff
@@ -472,12 +488,7 @@ def _request(self, method, url, body=None):
             else:
                 request = Request(url, data=body.encode('utf-8'), method=method)
 
-            request.add_header('Accept', 'application/json')
-            request.add_header('Content-Type', 'application/json;charset=UTF-8')
-
-            if parsed_url.username:
-                base64string = base64.b64encode('{0.username}:{0.password}'.format(parsed_url).encode())
-                request.add_header('Authorization', 'Basic {}'.format(base64string).decode())
+            request.add_remote_connection_headers(parsed_url)
```

Comment:
I like the idea of splitting this out. We could take it a step further though and remove the duplication from lines 450-458. Perhaps a `get_remote_connection_headers` that takes the URL and a boolean for `keep_alive`? It's also fine if you want to push this to a later date.